packages/strapi-design-system/src/Dialog/Dialog.tsx

```diff
@@ -29,7 +28,14 @@ const DialogHeader = styled(Flex)`
   border-bottom: 1px solid ${({ theme }) => theme.colors.neutral150};
 `;
 
-export const Dialog = ({ onClose, title, as, isOpen, id, ...props }) => {
+export interface DialogProps extends Omit<BoxProps, 'id'> {
+  onClose: () => void;
+  title: string;
+  isOpen: boolean;
+  id?: string | number | undefined;
+}
```

Comment:
```suggestion
export interface DialogProps extends BoxProps {
  onClose: () => void;
  title: string;
  isOpen: boolean;
}
```

I know you were following the PropTypes, but lets just do this for simplicity 😄